app/controllers/formbuilder/edit/code.js

```diff
@@ -36,12 +38,40 @@ export default class FormbuilderEditCodeController extends Controller {
       return;
     }
 
+    this.consoleValidateCode(newCode);
+
+    // The newCode is not assigned to this.fromCode as than the editor
+    // loses focus as you are udpating the content in the editor.
+    // Keeping the changes in another variable and at the end assigning
+    // the formCode to the updated code
+    this.formCodeUpdates = newCode;
+    this.model.handleCodeChange(this.formCodeUpdates);
+  });
+
+  @action
+  toggleOpenCloseConsoleState() {
+    this.consoleClosed = !this.consoleClosed;
+  }
+
+  async consoleValidateCode(newCode) {
     const shaclReport = await shaclValidateTtlCode(newCode);
     console.warn(
       'Formatted SHACL report: ',
       formatShaclValidationReport(shaclReport)
     );
 
+    this.warnings = [];
+
+    const formattedReport = formatShaclValidationReport(shaclReport);
+    formattedReport.errorDetails.forEach((error) => {
+      this.warnings.push({
+        subject: error.subject,
+        message: error.messages,
+      });
+    });
```

Comment:
We are only using warnings now in our shapes so this is ok. We should consider updating this that it checks the severity type of the error so it automatically can show errors and warnings. This can be done once we add the synstax errors etc..